libsql-sys/src/wal/sqlite3_wal.rs

```diff
@@ -170,15 +179,20 @@ impl Wal for Sqlite3Wal {
         }
     }
 
-    fn find_frame(&mut self, page_no: u32) -> Result<u32> {
+    fn find_frame(&mut self, page_no: u32) -> Result<Option<u32>> {
         let mut out: u32 = 0;
         let rc = unsafe {
-            (self.inner.methods.xFindFrame.unwrap())(self.inner.pData, page_no, &mut out as *mut _)
+            (self.inner.methods.xFindFrame.unwrap())(self.inner.pData, page_no, &mut out)
         };
+
         if rc != 0 {
             Err(Error::new(rc))
         } else {
-            Ok(out as _)
+            if out == 0 {
+                Ok(None)
+            } else {
+                Ok(Some(out as _))
+            }
```

Comment:
I don't follow the logic change here. Zero was never a valid frame number, was it? So is it about being more type-safe, and representing frame 0 as `None` instead of `u32`? If that's the case, maybe better to go full steam and implement it as `Option<std::num::NonZeroU32>`? Or is there another reason, and somehow both `Ok(0)` and `None` are reachable?

And a bonus question - why was `Option` needed in the first place?

Reply:
0 is used by SQLite to signify that a frame is not in the wal, it is more rust idiomatic to represent that by an option

Reply:
I changed that because it bit me, when the page 1 was not in the wal, and I was returned 0. I needed to go the the source to understand that 0 was not a valid index

Reply:
right - so `Option<NonZeroU32>` is a nice pick, it's guaranteed to just be represented as an u32 with 0 optimized to `None`: https://doc.rust-lang.org/beta/core/num/struct.NonZeroU32.html#layout-1